tests/unit/dataSource/remote/tests.html

```diff
@@ -406,10 +406,58 @@
 			//ds.settings.filtering.exprString = "";
 			ds.settings.paging.enabled = false;
 			// compare the "params" string with the expected one 
-			var params = $.param(ds._encodeUrl());
+			params = $.param(ds._encodeUrl());
 
 			equal(params, "%24filter=startswith(tolower(ProductID)%2C'a')%20eq%20true%20and%20endswith(tolower(Name)%2C'b')%20eq%20true%20and%20indexof(tolower(Price)%2C'')%20ge%200%20and%20indexof(tolower(ProductID)%2C'c')%20eq%20-1%20and%20tolower(Name)%20eq%20'1'%20and%20Name%20eq%201%20and%20tolower(Name)%20ne%20'abc'%20and%20Name%20gt%205%20and%20Name%20lt%202%20and%20Name%20ge%204%20and%20Name%20le%207", "%24filter=startswith(tolower(ProductID)%2C'a')%20eq%20true%20and%20endswith(tolower(Name)%2C'b')%20eq%20true%20and%20indexof(tolower(Price)%2C'')%20ge%200%20and%20indexof(tolower(ProductID)%2C'c')%20eq%20-1%20and%20tolower(Name)%20eq%20'1'%20and%20Name%20eq%201%20and%20tolower(Name)%20ne%20'abc'%20and%20Name%20gt%205%20and%20Name%20lt%202%20and%20Name%20ge%204%20and%20Name%20le%207");
 			
+			ds.settings.filtering.expressions = [
+				{fieldName: "ProductID", cond: "null", logic: "or"}, 
+				{fieldName: "Name", cond: "notNull", logic: "or"},
+				{fieldName: "Price", cond: "empty", logic: "or"}, 
+				{fieldName: "ProductID", cond: "notEmpty", logic: "or"},
+				{fieldName: "ReleaseDate", cond: "on", expr:  new Date(0), logic: "or"},
+				{fieldName: "ReleaseDate", cond: "notOn", expr:  new Date(0), logic: "or"},
+				{fieldName: "ReleaseDate", cond: "after", expr:  new Date(0), logic: "or"},
+				{fieldName: "ReleaseDate", cond: "before", expr:  new Date(0), logic: "or"},
+				{fieldName: "ReleaseDate", cond: "today", expr:  new Date(0), logic: "or"},
+				{fieldName: "ReleaseDate", cond: "yesterday", expr:  new Date(0)}
+			];
+			//ds.settings.filtering.exprString = "";
+			ds.settings.paging.enabled = false;
+			// compare the "params" string with the expected one 
+			params = $.param(ds._encodeUrl());
+			equal(params, "%24filter=ProductID%20eq%20null%20or%20Name%20ne%20null%20or%20length(Price)%20eq%200%20or%20length(ProductID)%20gt%200%20or%20day(ReleaseDate)%20eq%201%20and%20month(ReleaseDate)%20eq%201%20and%20year(ReleaseDate)%20eq%201970%20or%20day(ReleaseDate)%20ne%201%20or%20month(ReleaseDate)%20ne%201%20or%20year(ReleaseDate)%20ne%201970%20or%20ReleaseDate%20gt%20DateTime'1970-01-01T23%3A59%3A59'%20or%20ReleaseDate%20lt%20DateTime'1970-01-01'%20or%20day(ReleaseDate)%20eq%201%20and%20month(ReleaseDate)%20eq%201%20and%20year(ReleaseDate)%20eq%201970%20or%20day(ReleaseDate)%20eq%201%20and%20month(ReleaseDate)%20eq%201%20and%20year(ReleaseDate)%20eq%201970&filterLogic=or");;
```

Comment:
I don't see any asserts after this line. Probably you're missing an equal check.